appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/DOLUtils.java

```diff
@@ -1009,7 +998,9 @@ public static String getComponentEnvId(JndiNameEnvironment env) {
      * @return true if the class is white-listed
      */
     public static boolean isWhiteListed(Application application, String className) {
-        for (String packageName : Iterables.concat(application.getWhitelistPackages(), SYSTEM_PACKAGES)) {
+        List<String> packages = Arrays.asList(SYSTEM_PACKAGES);
+        packages.addAll(application.getWhitelistPackages());
```

Comment:
Building the union list seems unnecessary since we could just do the check on the two source lists as well.